doc/reference/kernel/other/fatal.rst

```diff
@@ -0,0 +1,245 @@
+	[00:00:00.000,000] <err> os: r0/a1:  0x00000004  r1/a2:  0x0000016f  r2/a3:  0x00000000
+	[00:00:00.000,000] <err> os: r3/a4:  0x00000000 r12/ip:  0x00000000 r14/lr:  0x00000a6d
+	[00:00:00.000,000] <err> os:  xpsr:  0x61000000
+	[00:00:00.000,000] <err> os: Faulting instruction address (r15/pc): 0x00009fe4
+	[00:00:00.000,000] <err> os: >>> ZEPHYR FATAL ERROR 4: Kernel panic
+	[00:00:00.000,000] <err> os: Current thread: 0x20000414 (main)
+	[00:00:00.000,000] <err> os: Halting system
+
+__ASSERT_EVAL()
+---------------
+
+The ``__ASSERT_EVAL()`` macro can also be used inside kernel and application
+code, with special semantics for the evalation of its arguments.
+
+It makes use of the ``__ASSERT()`` macro, but has some extra flexibility. It
+allows the developer to specify different actions depending whether the
+``__ASSERT()`` macro is enabled or not.  This can be particularly useful to
+prevent the compiler from generating comments (errors, warnings or remarks)
+about variables that are only used with ``__ASSERT()`` being assigned a value,
+but otherwise unused when the ``__ASSERT()`` macro is disabled.
+
+Consider the following example:
+
+.. code-block:: c
+
+  int x;
+  x = foo();
+  __ASSERT(x != 0, "foo() returned zero!");
+
+If ``__ASSERT()`` is disabled, then 'x' is assigned a value, but never used.
+This type of situation can be resolved using the __ASSERT_EVAL() macro.
+
+.. code-block:: c
+
+  __ASSERT_EVAL ((void) foo(),
+  		 int x = foo(),
+                 x != 0,
+                 "foo() returned zero!");
+
+The first parameter tells ``__ASSERT_EVAL()`` what to do if ``__ASSERT()`` is
+disabled.  The second parameter tells ``__ASSERT_EVAL()`` what to do if
+``__ASSERT()`` is enabled.  The third and fourth parameters are the parameters
+it passes to ``__ASSERT()``.
+
+__ASSERT_NO_MSG()
+-----------------
+
+The ``__ASSERT_NO_MSG()`` macro can be used to perform an assertion that
+reports the failed test and its location, but lacks additional debugging
+information provided to assist the user in diagnosing the problem; its use is
+discouraged.
+
+Build Assertions
+================
+
+Zephyr provides two macros for performing build-time assertion checks.
+These are evaluated completely at compile-time, and are always checked.
+
+BUILD_ASSERT_MSG()
+------------------
+
+This has the same semantics as C's ``_Static_assert`` or C++'s
+``static_assert``. If the evaluation fails a build error will be generated by
+the compiler. If the compiler supports it, the provided message will be printed
+to provide further context.
+
+Unlike ``__ASSERT()``, the message must be a static string, without
+``printf()``-like format codes or extra arguments.
+
+For example, suppose this check fails:
+
+.. code-block:: c
+
+	BUILD_ASSERT_MSG(FOO == 2000,
+			 "Invalid value of FOO");
+
+With GCC, the output resembles:
+
+.. code-block:: none
+
+	/home/apboie/projects/zephyr3/zephyr/tests/kernel/fatal/src/main.c: In function 'test_main':
```

Comment:
very minor: Maybe you want to chop a bit the path, given that it is just an example of the error it may as well have a few less folders.